### ISSUE TITLE
PyPi pip install does not include all depencencies (scipy, matplotlib)

### ISSUE THREAD

**Issue by GenevieveBuckley** [scikit-image/scikit-image#3887]
## Description
I notice that if I create an empty conda environment then pip install scikit-image (and nothing else), an error occurs when you try to import skimage because it's missing dependencies scipy and matplotlib. 

I assume this is not the intended behaviour, and it doesn't happen if I install scikit-image from conda-forge.

## Way to reproduce

```
conda create -n skimage pip
conda activate skimage
pip install scikit-image --no-cache-dir
python
>>> import skimage
...
ModuleNotFoundError: No module named 'scipy'
```

<details>
  <summary>See details of the environment creation & scikit-image installation</summary>

```
(base) C:\Users\genevieb>conda create -n skimage pip
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.12
  latest version: 4.6.14

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage

  added / updated specs:
    - pip


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.3.9           |           py37_0         155 KB

The following NEW packages will be INSTALLED:

    ca-certificates: 2019.1.23-0
    certifi:         2019.3.9-py37_0
    openssl:         1.1.1b-he774522_1
    pip:             19.1-py37_0
    python:          3.7.3-h8c8aaf0_1
    setuptools:      41.0.1-py37_0
    sqlite:          3.28.0-he774522_0
    vc:              14.1-h0510ff6_4
    vs2015_runtime:  14.15.26706-h3a45250_4
    wheel:           0.33.1-py37_0
    wincertstore:    0.2-py37_0

Proceed ([y]/n)? y


Downloading and Extracting Packages
certifi-2019.3.9     | 155 KB    | ######################################################## | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate skimage
#
# To deactivate an active environment, use
#
#     $ conda deactivate


(base) C:\Users\genevieb>conda activate skimage

(skimage) C:\Users\genevieb>which pip
/c/Users/genevieb/AppData/Local/conda/conda/envs/skimage/Scripts/pip

(skimage) C:\Users\genevieb>pip install scikit-image --no-cache-dir
Collecting scikit-image
  Downloading https://files.pythonhosted.org/packages/79/16/c5a36a03f90d4a246791d4ff1879f1868e1c5db58fac9f03427395c5e2d6/scikit_image-0.15.0-cp37-none-win_amd64.whl (25.6MB)
     |████████████████████████████████| 25.6MB 6.4MB/s
Collecting PyWavelets>=0.4.0 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/71/57/1b6c06f2b681975e5b43ff13ba127031f668d65b592cc56fae6445b675c7/PyWavelets-1.0.3-cp37-none-win_amd64.whl (4.2MB)
     |████████████████████████████████| 4.2MB 6.8MB/s
Collecting imageio>=2.0.1 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/af/0a/943c965d372dae0b1f1482677d29030ab834351a61a9a632fd62f27f1523/imageio-2.5.0-py3-none-any.whl (3.3MB)
     |████████████████████████████████| 3.3MB 6.4MB/s
Collecting pillow>=4.3.0 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/40/f2/a424d4d5dd6aa8c26636969decbb3da1c01286d344e71429b1d648bccb64/Pillow-6.0.0-cp37-cp37m-win_amd64.whl (2.0MB)
     |████████████████████████████████| 2.0MB ...
Collecting networkx>=2.0 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/85/08/f20aef11d4c343b557e5de6b9548761811eb16e438cee3d32b1c66c8566b/networkx-2.3.zip (1.7MB)
     |████████████████████████████████| 1.8MB ...
Collecting numpy>=1.9.1 (from PyWavelets>=0.4.0->scikit-image)
  Downloading https://files.pythonhosted.org/packages/4e/9d/c129d78e6b942303b762ccfdf1f8339de80c5e6021b14ef0c99ec5bdc6aa/numpy-1.16.3-cp37-cp37m-win_amd64.whl (11.9MB)
     |████████████████████████████████| 11.9MB ...
Collecting decorator>=4.3.0 (from networkx>=2.0->scikit-image)
  Downloading https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Building wheels for collected packages: networkx
  Building wheel for networkx (setup.py) ... done
  Stored in directory: C:\Users\genevieb\AppData\Local\Temp\pip-ephem-wheel-cache-__wf98oo\wheels\de\63\64\3699be2a9d0ccdb37c7f16329acf3863fd76eda58c39c737af
Successfully built networkx
Installing collected packages: numpy, PyWavelets, pillow, imageio, decorator, networkx, scikit-image
Successfully installed PyWavelets-1.0.3 decorator-4.4.0 imageio-2.5.0 networkx-2.3 numpy-1.16.3 pillow-6.0.0 scikit-image-0.15.0

(skimage) C:\Users\genevieb>python
Python 3.7.3 (default, Apr 24 2019, 15:29:51) [MSC v.1915 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
```
</details>

And the full traceback is below:

```python
>>> import skimage
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\__init__.py", line 127, in <module>
    from .util.dtype import (img_as_float32,
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\util\__init__.py", line 12, in <module>
    from ._montage import montage
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\util\_montage.py", line 2, in <module>
    from .. import exposure
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\exposure\__init__.py", line 1, in <module>
    from .exposure import histogram, equalize_hist, \
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\exposure\exposure.py", line 3, in <module>
    from ..color import rgb2gray
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\color\__init__.py", line 1, in <module>
    from .colorconv import (convert_colorspace,
  File "C:\Users\genevieb\AppData\Local\conda\conda\envs\skimage\lib\site-packages\skimage\color\colorconv.py", line 55, in <module>
    from scipy import linalg
ModuleNotFoundError: No module named 'scipy'

>>> import matplotlib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'matplotlib'

>>> import scipy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'scipy'
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
>>> from __future__ import print_function
>>> import sys; print(sys.version)
3.7.3 (default, Apr 24 2019, 15:29:51) [MSC v.1915 64 bit (AMD64)]
>>> import platform; print(platform.platform())
Windows-10-10.0.17763-SP0
>>> import numpy; print("numpy version: {}".format(numpy.__version__))
numpy version: 1.16.3
```
And if I manually pip install the missing `scipy` and `matplotlib` dependencies, then I can run:
```python
>>> import skimage; print("scikit-image version: {}".format(skimage.__version__))
scikit-image version: 0.15.0
```

**Comment by jni** [comment]
Definitely seems like a bug to me! Thanks for the report!

**Comment by hmaarrfk** [comment]
can't really recreate (on linux).... strange do you have anything installed on your system python3?
did you activate an environment inside an environment?
I think there may are things some issues with maybe multiple paths and it is detecting scipy in one of them?

I will try on windows when I reboot.

<details> 

```bash
(sktest) mark2@xps ~
$ pip install scikit-image                                                                                                                                                                                      
Collecting scikit-image
  Downloading https://files.pythonhosted.org/packages/2e/21/ea56c8bb2e8112837dd71aebeb2ac67913e784911c0d7f493a593fa1a207/scikit_image-0.15.0-cp37-cp37m-manylinux1_x86_64.whl (26.3MB)
     |████████████████████████████████| 26.3MB 675kB/s 
Collecting PyWavelets>=0.4.0 (from scikit-image)
  Using cached https://files.pythonhosted.org/packages/e3/91/607449f27a3a068739498eddf921cf2686f0a71fa8b44a8efce0af9446a8/PyWavelets-1.0.3-cp37-cp37m-manylinux1_x86_64.whl
Collecting scipy>=0.17.0 (from scikit-image)
  Using cached https://files.pythonhosted.org/packages/3e/7e/5cee36eee5b3194687232f6150a89a38f784883c612db7f4da2ab190980d/scipy-1.2.1-cp37-cp37m-manylinux1_x86_64.whl
Collecting matplotlib!=3.0.0,>=2.0.0 (from scikit-image)
  Using cached https://files.pythonhosted.org/packages/83/2a/e47bbd9396af32376863a426baed62d9bf3091f81defd1fe81c5f33b11a3/matplotlib-3.0.3-cp37-cp37m-manylinux1_x86_64.whl
Collecting imageio>=2.0.1 (from scikit-image)
  Using cached https://files.pythonhosted.org/packages/af/0a/943c965d372dae0b1f1482677d29030ab834351a61a9a632fd62f27f1523/imageio-2.5.0-py3-none-any.whl
Collecting networkx>=2.0 (from scikit-image)
Collecting pillow>=4.3.0 (from scikit-image)
  Using cached https://files.pythonhosted.org/packages/c1/e6/ce127fa0ac17775bc7887c432ffe945c49ae141f01b477b7cd5e63b16bb5/Pillow-6.0.0-cp37-cp37m-manylinux1_x86_64.whl
Collecting numpy>=1.9.1 (from PyWavelets>=0.4.0->scikit-image)
  Using cached https://files.pythonhosted.org/packages/bb/76/24e9f32c78e6f6fb26cf2596b428f393bf015b63459468119f282f70a7fd/numpy-1.16.3-cp37-cp37m-manylinux1_x86_64.whl
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 (from matplotlib!=3.0.0,>=2.0.0->scikit-image)
  Using cached https://files.pythonhosted.org/packages/dd/d9/3ec19e966301a6e25769976999bd7bbe552016f0d32b577dc9d63d2e0c49/pyparsing-2.4.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib!=3.0.0,>=2.0.0->scikit-image)
  Downloading https://files.pythonhosted.org/packages/93/f8/518fb0bb89860eea6ff1b96483fbd9236d5ee991485d0f3eceff1770f654/kiwisolver-1.1.0-cp37-cp37m-manylinux1_x86_64.whl (90kB)
     |████████████████████████████████| 92kB 488kB/s 
Collecting python-dateutil>=2.1 (from matplotlib!=3.0.0,>=2.0.0->scikit-image)
  Using cached https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl
Collecting cycler>=0.10 (from matplotlib!=3.0.0,>=2.0.0->scikit-image)
  Using cached https://files.pythonhosted.org/packages/f7/d2/e07d3ebb2bd7af696440ce7e754c59dd546ffe1bbe732c8ab68b9c834e61/cycler-0.10.0-py2.py3-none-any.whl
Collecting decorator>=4.3.0 (from networkx>=2.0->scikit-image)
  Using cached https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in /home/mark2/miniconda3/envs/sktest/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib!=3.0.0,>=2.0.0->scikit-image) (41.0.1)
Collecting six>=1.5 (from python-dateutil>=2.1->matplotlib!=3.0.0,>=2.0.0->scikit-image)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Installing collected packages: numpy, PyWavelets, scipy, pyparsing, kiwisolver, six, python-dateutil, cycler, matplotlib, pillow, imageio, decorator, networkx, scikit-image
Successfully installed PyWavelets-1.0.3 cycler-0.10.0 decorator-4.4.0 imageio-2.5.0 kiwisolver-1.1.0 matplotlib-3.0.3 networkx-2.3 numpy-1.16.3 pillow-6.0.0 pyparsing-2.4.0 python-dateutil-2.8.0 scikit-image-0.15.0 scipy-1.2.1 six-1.12.0
```
</details>

**Comment by hmaarrfk** [comment]
whoa!!! it does bug out on windows:


<details>

```
(base) C:\Users\mark>conda activate sktest

(sktest) C:\Users\mark>pip install scikit-image
Collecting scikit-image
  Downloading https://files.pythonhosted.org/packages/79/16/c5a36a03f90d4a246791d4ff1879f1868e1c5db58fac9f03427395c5e2d6/scikit_image-0.15.0-cp37-none-win_amd64.whl (25.6MB)
     |████████████████████████████████| 25.6MB 1.1MB/s
Collecting networkx>=2.0 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/85/08/f20aef11d4c343b557e5de6b9548761811eb16e438cee3d32b1c66c8566b/networkx-2.3.zip (1.7MB)
     |████████████████████████████████| 1.8MB 1.3MB/s
Collecting pillow>=4.3.0 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/40/f2/a424d4d5dd6aa8c26636969decbb3da1c01286d344e71429b1d648bccb64/Pillow-6.0.0-cp37-cp37m-win_amd64.whl (2.0MB)
     |████████████████████████████████| 2.0MB 1.7MB/s
Collecting imageio>=2.0.1 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/af/0a/943c965d372dae0b1f1482677d29030ab834351a61a9a632fd62f27f1523/imageio-2.5.0-py3-none-any.whl (3.3MB)
     |████████████████████████████████| 3.3MB 1.1MB/s
Collecting PyWavelets>=0.4.0 (from scikit-image)
  Downloading https://files.pythonhosted.org/packages/71/57/1b6c06f2b681975e5b43ff13ba127031f668d65b592cc56fae6445b675c7/PyWavelets-1.0.3-cp37-none-win_amd64.whl (4.2MB)
     |████████████████████████████████| 4.2MB 1.6MB/s
Collecting decorator>=4.3.0 (from networkx>=2.0->scikit-image)
  Downloading https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Collecting numpy (from imageio>=2.0.1->scikit-image)
  Downloading https://files.pythonhosted.org/packages/4e/9d/c129d78e6b942303b762ccfdf1f8339de80c5e6021b14ef0c99ec5bdc6aa/numpy-1.16.3-cp37-cp37m-win_amd64.whl (11.9MB)
     |████████████████████████████████| 11.9MB 1.7MB/s
Building wheels for collected packages: networkx
  Building wheel for networkx (setup.py) ... done
  Stored in directory: C:\Users\mark\AppData\Local\pip\Cache\wheels\de\63\64\3699be2a9d0ccdb37c7f16329acf3863fd76eda58c39c737af
Successfully built networkx
Installing collected packages: decorator, networkx, pillow, numpy, imageio, PyWavelets, scikit-image
Successfully installed PyWavelets-1.0.3 decorator-4.4.0 imageio-2.5.0 networkx-2.3 numpy-1.16.3 pillow-6.0.0 scikit-image-0.15.0
```

</details>

**Comment by hmaarrfk** [comment]
The metadata.json file is clearly missing scipy in it

```
  "requires": "numpy (>= 1.11)", 
  "run_requires": [
    {
      "requires": ["PyWavelets (>=0.4.0)", "imageio (>=2.0.1)", "networkx (>=2.0)", "pillow (>=4.3.0)"]
    }, 
    { 
      "requires": ["SimpleITK", "astropy (>=1.2.0)", "cloudpickle (>=0.2.1)", "dask[array] (>=0.9.0)", "pyamg", "qtpy", "tifffile"], 
      "extra": "optional"
    }, 
    {
      "requires": ["cloudpickle (>=0.2.1)", "dask[array] (>=0.9.0)", "numpydoc (>=0.6)", "pytest-runner", "scikit-learn", "sphinx (>=1.3,!=1.7.8)", "sphinx-copybutton", "sphinx-gallery"], 
      "extra": "docs"
    }, 
    {
      "requires": ["codecov", "flake8", "pytest (!=3.7.3)", "pytest-cov", "pytest-faulthandler", "pytest-localserver"], 
      "extra": "test"
    }
  ], 
  "requires_python": ">=3.5", 
  "version": "0.15.0"
}
```

**Comment by hmaarrfk** [comment]
Apparently we don't even need the metadata.json file in the new wheel format as of 0.30.0

> Dropped DESCRIPTION.rst and metadata.json from the list of generated files

https://github.com/pypa/wheel/blob/master/docs/news.rst

we can probably take it away.

**Comment by hmaarrfk** [comment]
I think we can remove the wheel pinning
https://github.com/scikit-image/scikit-image-wheels/blob/2ae9984071d37c4b5c28f2ea384ead6184a0b4dd/.appveyor.yml#L56

isn't wheel included in pip now by default?

**Comment by GenevieveBuckley** [comment]
> whoa!!! it does bug out on windows:

Thanks for investigating 😄

**Comment by matthew-brett** [comment]
@hmaarrfk - the reason for the wheel pinning is in the comment in the line above - later version of wheel insert specific ABI platform tags that weren't compatible with some earlier pip versions.  But yes, that can go now.

**Comment by hmaarrfk** [comment]
Thanks for confirming @matthew-brett 

This PR
https://github.com/scikit-image/scikit-image-wheels/pull/19
should fix things by unpinning dependencies.

**Comment by soupault** [comment]
@hmaarrfk did you get a chance to verify the fix?

**Comment by hmaarrfk** [comment]
Yeah i think the fix was to unpin wheel is correct. I can't seem to find nightly builds for windows, so it is difficult to test.

**Comment by sciunto** [comment]
@NelleV Can you check that this will be fixed by 0.16 please?

**Comment by sciunto** [comment]
I suspect it is fixed. Please, reopen if needed.

**Comment by GenevieveBuckley** [comment]
Can confirm, is fixed with scikit-image 0.16.1 (pip installed on Windows 10).
